src/rendering/RenderUtil.cc

```diff
@@ -400,6 +400,20 @@ void RenderUtil::Update()
     {
       this->dataPtr->sceneManager.CreateLight(
           std::get<0>(light), std::get<1>(light), std::get<2>(light));
+
+      // create a new id for the light visual
+      auto timeout = 100000u;
+      for (auto i = 0u; i < timeout; ++i)
```

Comment:
I don't think it's a good idea to block the render thread. If this may take time to be completed, I suggest we queue it and process it when possible.

Reply:
the right name for this variable is `attempts`, this loop has no relation with time or sleep. It's looking for an available entity ID.

Reply:
Ohh ok understood! I wonder if there's a chance of entity ID collision when new entities are inserted into the simulation from the backend. I didn't notice problems in my manual testing earlier.

Reply:
In this method I can't access the ECM to call `CreateEntity`. Is there any other way to get an empty ID ?

Reply:
For the entity preview when spawning, there's this method in the `Scene3D` plugin:

https://github.com/ignitionrobotics/ign-gazebo/blob/766bb23eaca85ff60941e2ef852c4f286356fbcc/src/gui/plugins/scene3d/Scene3D.cc#L927-L937

That plugin isn't accessible from `RenderUtil`. But I think it could make sense to move the light visual logic from `RenderUtil` to `Scene3d`. The light visuals are only supposed to show on the client scene, not on the server. But `RenderUtil` is used by both scenes. I think this would be a big change and it may take longer to merge this PR though :grimacing:

Reply:
I can open an issue and merge this PR. Waht do you think @chapulina ?

Reply:
friendly ping @chapulina

Reply:
An issue sounds good, thanks!

Reply:
issue created https://github.com/ignitionrobotics/ign-gazebo/issues/633